### PR TITLE
Legitimately Fixes The CtrlGrab Bug Now

### LINKED ISSUE #21927: Quickgrabbing (ctrl+click spam) not fixed by latest PR
Exactly what it says on the tin.

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
I'll be like Mekhi and have 5 prs for the same fix now.

This works up until you start tightening the grip, to which you can spam [in vain since it will just stop your grab]. I'll probably fix that later if I can figure out what's going on with it.

Oddly enough, I had to use `next_click` instead of `changeNext_move`, kinda dumb but whatever.

Fixes #21927

### CODE REVIEW

**Comment by Iamgoofball** [comment]
how is this a bug

**Comment by ExcessiveUseOfCobblestone** [comment]
because there's no cooldown on Ctrl+click to grab like there is for going to grab intent and normally clicking. If you need to see it ingame I can show you.

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -228,15 +228,14 @@
...
-        H.dna.species.grab(H, src, H.martial_art)
-        H.changeNext_move(CLICK_CD_MELEE)
-        return TRUE
-    return ..()
-
+	if(ishuman(user) && Adjacent(user))
+		var/mob/living/carbon/human/H = user
+		H.dna.species.grab(H, src, H.martial_art)
+		H.next_click = world.time + CLICK_CD_MELEE
+	return
```
Comment (optimumtact) [inline code/_onclick/click.dm]: why are you not calling the parent anymore, that means they wont' get a message

Comment (optimumtact) [inline code/_onclick/click.dm]: This seems really dodgy and hacky to me

**Comment by jughu** [comment]
if we have cooldowns can we remove the grab resist then since its stupid

**Comment by kevinz000** [comment]
im insulted :(

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -228,15 +228,14 @@
 	var/mob/living/ML = user
 	if(istype(ML))
 		ML.pulled(src)
+	return
```
Comment (ChangelingRain) [inline code/_onclick/click.dm]: trailing return!

File: code/_onclick/click.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/_onclick/click.dm]: also this is a trailing return!

**Comment by ExcessiveUseOfCobblestone** [comment]
@jughu you already have cooldowns, this just makes them consistent.

**Comment by optimumtact** [comment]
It was a mistake to look into this, I see the problem, which is that next move only applies to the clickon proc and not alternate clicks with modifiers.

However it made me realise we need a system to handle click cooldowns for different click types and that made me sad because it means I have to look at click code

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm sorry I did this to you citrus.

**Pull request merged**